src/librbd/io/Types.h

```diff
@@ -11,6 +11,18 @@
 namespace librbd {
 namespace io {
 
+#define RBD_QOS_IOPS_THROTTLE				1 << 0
+#define RBD_QOS_BPS_THROTTLE				1 << 1
+#define RBD_QOS_READ_IOPS_THROTTLE			1 << 2
+#define RBD_QOS_WRITE_IOPS_THROTTLE			1 << 3
+#define RBD_QOS_READ_BPS_THROTTLE			1 << 4
+#define RBD_QOS_WRITE_BPS_THROTTLE			1 << 5
+
+#define RBD_QOS_BPS_MASK	(RBD_QOS_BPS_THROTTLE | RBD_QOS_READ_BPS_THROTTLE | RBD_QOS_WRITE_BPS_THROTTLE)
+#define RBD_QOS_IOPS_MASK	(RBD_QOS_IOPS_THROTTLE | RBD_QOS_READ_IOPS_THROTTLE | RBD_QOS_WRITE_IOPS_THROTTLE)
```

Comment:
Nit: used anywhere?

Reply:
Yup, such as
https://github.com/ceph/ceph/pull/21635/files#diff-82401c66599ad9652ae4be08000e5814R84

We need to check the flag with these mask to decide return ```spec->extents_length();``` (BPS tokens) or 1 (IOPS tokens).